lib/gringotts/gateways/monei.ex

```diff
@@ -178,22 +176,18 @@ defmodule Gringotts.Gateways.Monei do
 
   The following session shows how one would (pre) authorize a payment of $40 on a sample `card`.
 
-      iex> opts = [currency: "EUR"] # The default currency is EUR, and this is just for an example.
+      iex> amount = %{amount: 40, currency: "EUR"}
       iex> card = %Gringotts.CreditCard{first_name: "Jo", last_name: "Doe", number: "4200000000000000", year: 2099, month: 12, verification_code:  "123", brand: "VISA"}
-      iex> auth_result = Gringotts.authorize(Gringotts.Gateways.Monei, 40, card, opts)
+      iex> auth_result = Gringotts.authorize(Gringotts.Gateways.Monei, amount, card, opts)
       iex> auth_result.id # This is the authorization ID
   """
-  @spec authorize(number, CreditCard.t(), keyword) :: {:ok | :error, Response}
-  def authorize(amount, card = %CreditCard{}, opts) when is_integer(amount) do
-    authorize(amount / 1, card, opts)
-  end
-
-  def authorize(amount, card = %CreditCard{}, opts) when is_float(amount) do
+  @spec authorize(Money.t, CreditCard.t(), keyword) :: {:ok | :error, Response}
+  def authorize(amount, card = %CreditCard{}, opts) do
     params =
       [
         paymentType: "PA",
-        amount: :erlang.float_to_binary(amount, decimals: 2),
-        currency: currency(opts)
+        amount: amount |> Money.amount |> Decimal.to_float |> :erlang.float_to_binary(decimals: 2),
```

Comment:
We could rename the protocol method `Money.amount` to `Money.value` perhaps, then this would read:
```elixir
amount: amount |> Money.value |> Decimal.to_float |> :erlang.float_to_binary(decimals: 2)
```

Writing `Money.value(amount)` is better than writing `Money.amount(amount)`, no?

Reply:
LGTM 👍